plugins/source/postgresql/client/sync.go

```diff
@@ -141,7 +142,17 @@ func syncTable(ctx context.Context, tx pgx.Tx, table *schema.Table, res chan<- m
 
 			scalar.AppendToBuilder(builder.Field(i), s)
 		}
-		res <- &message.SyncInsert{Record: builder.NewRecord()} // NewRecord resets the builder for reuse
+
+		processed++
+		if processed%c.pluginSpec.RowsPerRecord == 0 {
```

Comment:
nit: I think we should either not reset `processed` to 0 or use `processed >= c.pluginSpec.RowsPerRecord` here, but not both, it sends mixed messages of what should be happening